### ISSUE TITLE
RichTextEditor cycle in updateUIView

### ISSUE THREAD

**Issue by DominikBucher12** [danielsaidi/RichTextKit#56]
Hello,

The following code in `RichTextEditor` causes swift warning `Publishing changes from within view updates is not allowed, this will cause undefined behavior.` Also because we update UI from within updateUI function, it causes infinite loop and does not allow to get past the point.

```   
public func updateUIView(_ view: UIViewType, context: Context) {
        textView.attributedString = text.wrappedValue
        richTextContext.highlightedRange = nil // This causes cyclic update to UIView 
    }
```

https://github.com/danielsaidi/RichTextKit/assets/17381941/2431bbfd-8cb1-4e83-be21-1c698bd3bf33

# Workaround:
Delete the assignment of `highlightedRange`

Xcode Version: 14.3 (14E222b)
iOS 14 Simulator (or any other)

**Comment by danielsaidi** [comment]
I noticed this too. The new release adds a dispatch queue main async around it, but I think it may still stutter a bit. I will investigate and see if we need a patch update.

**Comment by danielsaidi** [comment]
Hi again @DominikBucher12 

I could reproduce the error and have removed all code from the update functions.

The bug fix is available in the 0.6.1 patch version.

Thanks for notifying me! 👍
